zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/OrchestratorZoneService.java

```diff
@@ -88,13 +171,305 @@ private ConnectionDetails buildConnectionDetails(String zoneName, IdentityZone i
+                                                       "subdomain : " + subdomain + " Zone Id : " + zoneId + " Exception is :" + e.getMessage());
+        }
+    }
+
+    private void createDefaultIdp(IdentityZone created) throws OrchestratorZoneServiceException {
+        try {
+            IdentityProvider defaultIdp = new IdentityProvider();
+            defaultIdp.setName(OriginKeys.UAA);
+            defaultIdp.setType(OriginKeys.UAA);
+            defaultIdp.setOriginKey(OriginKeys.UAA);
+            defaultIdp.setIdentityZoneId(created.getId());
+            UaaIdentityProviderDefinition idpDefinition = new UaaIdentityProviderDefinition();
+            idpDefinition.setPasswordPolicy(null);
+            defaultIdp.setConfig(idpDefinition);
+            idpProvisioning.create(defaultIdp, created.getId());
+        } catch (Exception e) {
+            logger.error("Unable to create identity provider for subdomain : " + created.getSubdomain() + " Exception " +
+                         "is : " + e.getMessage());
+            throw new OrchestratorZoneServiceException("Unexpected exception while create identity provider for " +
+                                                       "subdomain : ." + created.getSubdomain() + " Exception is : " + e.getMessage());
+        }
+    }
+
+    private IdentityZone createIdentityZone(IdentityZone identityZone)
+        throws OrchestratorZoneServiceException {
+        IdentityZone created = null;
+        try {
+            created = zoneProvisioning.create(identityZone);
+        } catch (Exception e) {
+            logger.error("Unable to create identity zone for subdomain : " + identityZone.getSubdomain() + " Exception is : " + e.getMessage());
+            throw new OrchestratorZoneServiceException("Unexpected exception while create identity zone for " +
+                                                       "subdomain : ." + identityZone.getSubdomain() + " Exception is : " + e.getMessage());
+        }
+        return created;
+    }
+
+    private IdentityZone geIdentityZone(String subdomain, String name, String id, String zoneSigningKey) throws OrchestratorZoneServiceException {
+        IdentityZone identityZone = new IdentityZone();
+        identityZone.setId(id);
+        identityZone.setName(name);
+        identityZone.setSubdomain(subdomain);
+        setTokenPolicy(zoneSigningKey, identityZone);
+        setSamlConfig(subdomain, identityZone);
+        identityZone.getConfig().getLinks().getLogout().setWhitelist(createDeploymentSpecificLogoutWhiteList());
+        return identityZone;
+    }
+
+    private void setSamlConfig(String subdomain, IdentityZone identityZone) throws OrchestratorZoneServiceException {
+        try {
+            identityZone.getConfig().setSamlConfig(createSamlConfig(subdomain));
+        } catch (Exception e) {
+            logger.error("Unable to create saml config for subdomain: " + subdomain + " Exception is : " + e.getMessage());
```

Comment:
Log entire stack trace instead of printing only message. This will helpful for debug.

Reply:
Done. Thanks!